File: src/TypeFormatter/ConstructorTypeFormatter.ts
import { BaseType } from "../Type/BaseType.js";
import { ConstructorType } from "../Type/ConstructorType.js";
import { FunctionTypeFormatter } from "./FunctionTypeFormatter.js";

export class ConstructorTypeFormatter extends FunctionTypeFormatter {
    public supportsType(type: BaseType): boolean {
        return type instanceof ConstructorType;
    }
}

File: internal/formatter/constructor_type_formatter.go
package formatter

import (
	"github.com/vega/ts-json-schema-generator/internal/config"
	"github.com/vega/ts-json-schema-generator/internal/types"
)

// ConstructorTypeFormatter mirrors src/TypeFormatter/ConstructorTypeFormatter.ts,
// which subclasses FunctionTypeFormatter and only overrides supportsType.
type ConstructorTypeFormatter struct {
	*FunctionTypeFormatter
}

func NewConstructorTypeFormatter(childTypeFormatter TypeFormatter, functions config.FunctionOptions) *ConstructorTypeFormatter {
	return &ConstructorTypeFormatter{FunctionTypeFormatter: NewFunctionTypeFormatter(childTypeFormatter, functions)}
}

func (f *ConstructorTypeFormatter) SupportsType(t types.Type) bool {
	_, ok := t.(*types.ConstructorType)
	return ok
}
